cli/command/telemetry_utils.go

```diff
@@ -27,7 +28,7 @@ func BaseMetricAttributes(cmd *cobra.Command) attribute.Set {
 // Note: this should be the last func to wrap/modify the PersistentRunE/RunE funcs before command execution.
 //
 // can also be used for spans!
-func InstrumentCobraCommands(cmd *cobra.Command, mp metric.MeterProvider) {
+func (cli *DockerCli) InstrumentCobraCommands(cmd *cobra.Command, mp metric.MeterProvider) {
```

Comment:
similar to above, i don't think we need the whole cli struct here.

I think we should also consolidate our approach when it comes to passing a parameter or attaching a function to a struct. Here we're attaching to the dockerCli struct (which i think could probably be changes if/when we extract shared telemetry bits to a separate pkg), in the `startCobraCommandTimer` and `BaseMetricAttributes` funcs (the only places the dockerCli is needed in here) we pass it as a param instead.

Are we planning on using any other data from the struct other than the `isTerminal()` func?